src/backend/storage/ipc/dsm.c

```diff
@@ -164,8 +165,9 @@ dsm_postmaster_startup(PGShmemHeader *shim)
 		dsm_cleanup_for_mmap();
 
 	/* Determine size for new control segment. */
-	maxitems = PG_DYNSHMEM_FIXED_SLOTS
-		+ PG_DYNSHMEM_SLOTS_PER_BACKEND * MaxBackends;
+	maxitems = PG_DYNSHMEM_FIXED_SLOTS +
+			   PG_DYNSHMEM_SLOTS_PER_BACKEND * MaxBackends +
+			   NUM_SHARED_SNAPSHOT_SLOTS * 2;
```

Comment:
Why NUM_SHARED_SNAPSHOT_SLOTS * 2？

Reply:
`maxitems = PG_DYNSHMEM_FIXED_SLOTS + PG_DYNSHMEM_SLOTS_PER_BACKEND * MaxBackends;`
It is pg formula for dsm preallocation methods. In cbdb, we use dsm in cursor snapshot dump, sharedsnapshot sync, etc.  But never bump up the number. So roughly give more dsm segment size